dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/LayoutDataFetcher.java

```diff
@@ -0,0 +1,44 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotcms.rendering.velocity.viewtools.DotTemplateTool;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.templates.design.bean.TemplateLayout;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.Map;
+
+/**
+ * This DataFetcher returns a {@link Map} representing a {@link Template} associated to the originally
+ * requested {@link HTMLPageAsset}.
+ */
+public class LayoutDataFetcher implements DataFetcher<TemplateLayout> {
+    @Override
+    public TemplateLayout get(final DataFetchingEnvironment environment) throws Exception {
+        try {
+            final DotGraphQLContext context = environment.getContext();
+            final User user = context.getUser();
+            final Contentlet page = environment.getSource();
+            final String pageModeAsString = (String) context.getParam("pageMode");
+
+            final PageMode mode = PageMode.get(pageModeAsString);
+
+            final HTMLPageAsset pageAsset = APILocator.getHTMLPageAssetAPI()
+                    .fromContentlet(page);
+
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6084048)